internal/pkg/comm/creds.go

```diff
@@ -28,26 +29,50 @@ var (
 // NewServerTransportCredentials returns a new initialized
 // grpc/credentials.TransportCredentials
 func NewServerTransportCredentials(
-	serverConfig *tls.Config,
+	serverConfig *TLSConfig,
 	logger *flogging.FabricLogger) credentials.TransportCredentials {
-
 	// NOTE: unlike the default grpc/credentials implementation, we do not
 	// clone the tls.Config which allows us to update it dynamically
-	serverConfig.NextProtos = alpnProtoStr
+	serverConfig.lock.Lock()
+	defer serverConfig.lock.Unlock()
+	serverConfig.config.NextProtos = alpnProtoStr
 	// override TLS version and ensure it is 1.2
-	serverConfig.MinVersion = tls.VersionTLS12
-	serverConfig.MaxVersion = tls.VersionTLS12
+	serverConfig.config.MinVersion = tls.VersionTLS12
+	serverConfig.config.MaxVersion = tls.VersionTLS12
 	return &serverCreds{
 		serverConfig: serverConfig,
 		logger:       logger}
 }
 
 // serverCreds is an implementation of grpc/credentials.TransportCredentials.
 type serverCreds struct {
-	serverConfig *tls.Config
+	serverConfig *TLSConfig
 	logger       *flogging.FabricLogger
 }
 
+type TLSConfig struct {
+	config *tls.Config
+	lock   *sync.Mutex
+}
+
+func NewTLSConfig(config *tls.Config, lock *sync.Mutex) *TLSConfig {
+	return &TLSConfig{
+		config: config,
+		lock:   lock,
+	}
+}
+
+func (sc *TLSConfig) Config() *tls.Config {
```

Comment:
I suggest that the TLSConfig struct would govern the lifecycle of its inner fields and not delegate it to the user of the struct:

1. Let's return a copy upon Config()
2. Let's add a mutator method that just updates whatever we want, under the lock.

This removes the need to external locking completely and simplifies the usage of this struct.